hickle/lookup.py

```diff
@@ -20,168 +22,1548 @@
+   load_dataset functions and PyContainer objects, along with the 
+   'class_register' and 'exclude_register' tables. See the other loaders
+   for examples.
 
 """
 
 
 # %% IMPORTS
 # Built-in imports
-from ast import literal_eval
-from importlib import import_module
-from inspect import isclass
-from itertools import starmap
+import sys
+import warnings
+import types
+import re
+import weakref
+import os.path
+from importlib.util import find_spec, module_from_spec,spec_from_file_location,spec_from_loader
+from importlib import invalidate_caches
 
-# hickle import
-from hickle.helpers import get_mro_list
+# Package imports
+import collections
+import dill as pickle
+import numpy as np
+import h5py as h5
 
+# hickle imports
+from .helpers import PyContainer,not_dumpable,nobody_is_my_name,no_compression,NotHicklable
+from .loaders import optional_loaders, attribute_prefix
 
-# %% GLOBALS
-# Define dict of all acceptable types
-types_dict = {}
-
-# Define dict of all acceptable hickle types
-hkl_types_dict = {}
-
-# Define list of types that should never be sorted
-types_not_to_sort = []
-
-# Empty list of loaded loader names
-loaded_loaders = []
-
-# Define dict containing validation functions for ndarray-like objects
-ndarray_like_check_fns = {}
-
-# Define conversion dict of all acceptable dict key types
-dict_key_types_dict = {
-    b'str': literal_eval,
-    b'float': float,
-    b'bool': bool,
-    b'int': int,
-    b'complex': complex,
-    b'tuple': literal_eval,
-    b'NoneType': literal_eval,
-    }
+if sys.version_info[:2] <= (3,5): # pragma: nocover
+    # define ModuleNotFoundError on __builtins__ to ensure below code is working
+    setattr(sys.modules['builtins'],'ModuleNotFoundError',getattr(sys.modules['builtins'],'ModuleNotFoundError',ImportError))
 
+# %% GLOBALS
 
 # %% FUNCTION DEFINITIONS
-def load_nothing(h_node):
-    pass
 
 
-#####################
-# ND-ARRAY checking #
-#####################
+def load_nothing(h_node, base_type , py_obj_type): # pragma: nocover
+    """
+    loads nothing
+    """
+    return nobody_is_my_name
+
+def dump_nothing(py_obj, h_group, name, **kwargs): # pragma: nocover
+    """
+    dumps nothing
+    """
+    return nobody_is_my_name
 
-def check_is_ndarray_like(py_obj):
-    # Obtain the MRO of this object
-    mro_list = py_obj.__class__.mro()
+# h5py < 2.10 ref_dtype has to be explicitly created by # call to
+#
+#      special_dtype(ref=h5py.Reference) 
+#
+# h5py >= 2.10 provides already a ref_dtype where metadata is alreadyi
+# properly set. Both ways to define ref_dtype are equivalent for
+# h5py >= 2.10 for any other mimick definiton for ref_dtype of
+# h5py >= 2.10 
+if ( 
+    h5.version.version_tuple[0] > 2 or
+    ( h5.version.version_tuple[0] == 2 and h5.version.version_tuple[1] >= 10 )
+):
+    link_dtype = h5.ref_dtype
 
-    # Create a function map
-    func_map = map(ndarray_like_check_fns.get, mro_list)
+else: # pragma: nocover
```

Comment:
Should also be covered.

Reply:
In case support of h5py versions smaller than 2.10 is not necessary this can be solved by removing the else clause and not checking whether h5py >= 2.10 is used. In any other case it is the same problem above that i have not found a better method to tell coverage not report lines which can not be hit when h5py >= 2.10 is used. and vice versa. I do not have access to any system where h5py < 2.10 would be installed. So if you and @telegraphic are ok i would solve this by just assuming minimumm required h5py version is 2.10 and drop the else clause here

Reply:
Then set the minimum requirement to that.